libs/README.md

```diff
@@ -0,0 +1,375 @@
+and update the `LaunchpadState`.
+
+Components of the launchpad app have to listen to the changes
+of the state and be re-rendered if the state is changed.
+
+### Actions
+
+To send actions like: transfer funds, save settings, etc. we
+should send an event to the `tari://actions` stream of the
+following `Action` type:
+
+```rust
+pub enum Action {
+    Connect,
+    SaveSettings(LaunchpadConfig),
+    ///
+    TransferFunds(Amount, Recipient),
+    // etc.
+}
+```
+
+The first event `Connect` is the special event that asks to re-send
+the current state and should be sent when the main window is loaded
+and the app is ready for processing events.
+
+The event `SaveSettings` sends a new configuration to the engine when
+the user changes it and saves it on the settings screen.
+
+When the configuration is saved, the engine sends a delta
+`UpdateConfig` which means the config was applied, and all affected
+containers will be restarted.
+
+To check whether the settings are changed or not, it's enough to compare
+an instance from the state and the new configuration that exists in the settings screen.
+
+
+### UI refactoring proposal
+
+To use the new state, we should subscribe the frontend to events
+from the `tari://actions` and provide that state using the **Redux**.
+Ideally, the state should be available for all components, but
+a rendering of them should depend on changes in specific state fields.
+
+#### The cases
+
+There are examples below how we could replaces some API calls
+with that state.
+
+##### Node Identity
+
+###### Original
+
+A node identity is necessary to start a wallet container. In the original
+implementation we have to wait when the node container is started and
+call the `node_identity` method.
+
+```rust
+pub async fn node_identity() -> Result<BaseNodeIdentity, String>
+```
+
+###### With the state
+
+The node identity shouldn't be requested explicitly because the engine
+gets it using the inner GRPC client as soon as the base node container
+is available.
+
+It's still possible to share it with the state, but the UI shouldn't
+care about it for starting the wallet.
+
+##### Syncing progress
+
+###### Original
+
+There is a method to subscribe to progress events from the base node:
+
+```rust
+pub async fn base_node_sync_progress(app: AppHandle<Wry>) -> Result<(), String>
+```
+
+The frontend used it to get progress during onboarding process.
+
+###### With the state
+
+The state has the field `containers` that keeps the `TaskStatus` objects:
+
+```rust
+pub struct TaskState {
+    pub status: TaskStatus,
+    pub tail: VecDeque<String>,
+}
+```
+
+The task state contains the status of the container:
+
+```rust
+pub enum TaskStatus {
+    Inactive,
+    Pending,
+    Progress(TaskProgress),
+    Active,
+}
+```
+
+And to get to know the container is ready or not (in progress) we should
+get a value of the following field in the frontend:
+
+```rust
+state.containers["Base Node"].status
```

Comment:
is there a list of all the container names like this we use to get these states, or just all based off the `display_name` [here](https://github.com/tari-project/tari-launchpad/blob/5c92aef886470c30cefcb8617c9e4ffc9a76ffbe/backend/src/docker/models.rs#L224)?